connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -107,15 +107,17 @@ message Join {
   }
 }
 
-// Relation of type [[Union]], at least one input must be set.
-message Union {
+// Relation of type [[SetOperation]]
+message SetOperation {
```

Comment:
can we have 3 individual plans?

Reply:
Of course union supports `ALL`. Not sure why we would need three plans for this?

Reply:
Because catalyst has 3 plans for set operations, I think it's better to follow it in proto so that they can evolve separately.

To @grundprinzip : we do support `UNION ALL`, but the logical plan for `UNION` is `Distinct(Union(...))`. The `Union` operator does not have a `isAll` flag.

Reply:
After a second thought, if we treat the proto layer as a standard for general SQL query plans, it doesn't have to be consistent with catalyst. Then a single `SetOperation` LGTM.

Reply:
Note: Spark has a tricky optimization for a long chain of `Union`. e.g. `df1.union(df2).union(df3).union...`, Spark does not create a left-deep `Union` tree (which makes analyzer/optimizer run very slow), but just a single `Union` with many children. I'm not sure if we want to do this optimization in Spark Connect clients or not. But if we do, we need a special `Union` proto definition which takes a Seq of children, not just left and right.

Reply:
Ok maybe we keep the same message as `SetOperation` for now. When we face the need for further optimization on specific case, we can propose ways to help that.

Reply:
nvm, Spark Connect DataFrame operations combine parsed plan, not analyzed plan, so we don't need this early Union optimization. A single `SetOperation` should work.

Reply:
That would be great then!